service/metric/handler.go

```diff
@@ -11,8 +12,13 @@ type Server struct {
 	metrics http.Handler
 }
 
+var doOnce sync.Once
+
 // NewServer returns a new metrics server.
 func NewServer() *Server {
+	doOnce.Do(func() {
```

Comment:
This is needed to run `InitMetrics()` once, because otherwise, Prometheus will complain about metrics registration. It only happens if we run tests. In normal case, this code will be executed once anyway.
.
In Golang you can reach the same using `init()` function. The code int this function will be executed only once on the import. But this is somehow magic and I don't like to use `init()`.